src/js/tabs/motors.js

```diff
@@ -662,41 +685,46 @@ TABS.motors.initialize = function (callback) {
 
                 if (i < MOTOR_CONFIG.motor_count && (MOTOR_CONFIG.use_dshot_telemetry || MOTOR_CONFIG.use_esc_sensor)) {
 
-                    const MAX_INVALID_PERCENT = 100,
-                          MAX_VALUE_SIZE = 6;
+                    let rpmMotorValue = reduceTelemetryValueSize(MOTOR_TELEMETRY_DATA.rpm[i]);
 
-                    let rpmMotorValue = MOTOR_TELEMETRY_DATA.rpm[i];
-
-                    // Reduce the size of the value if too big
-                    if (rpmMotorValue > 999999) {
-                        rpmMotorValue = (rpmMotorValue / 1000000).toFixed(5 - (rpmMotorValue / 1000000).toFixed(0).toString().length) + "M";  
-                    }
-
-                    rpmMotorValue = rpmMotorValue.toString().padStart(MAX_VALUE_SIZE);
+                    rpmMotorValue = rpmMotorValue.toString().padStart(self.MAX_TELEMETRY_VALUE_SIZE);
                     let telemetryText = i18n.getMessage('motorsRPM', {motorsRpmValue: rpmMotorValue});
-
                     
                     if (MOTOR_CONFIG.use_dshot_telemetry) {
 
                         let invalidPercent = MOTOR_TELEMETRY_DATA.invalidPercent[i];
 
                         let classError = (invalidPercent > MAX_INVALID_PERCENT) ? "warning" : "";
-                        invalidPercent = (invalidPercent / 100).toFixed(2).toString().padStart(MAX_VALUE_SIZE);
+                        invalidPercent = (invalidPercent / 100).toFixed(2).toString().padStart(self.MAX_TELEMETRY_VALUE_SIZE);
 
                         telemetryText += "<br><span class='" + classError + "'>";
                         telemetryText += i18n.getMessage('motorsRPMError', {motorsErrorValue: invalidPercent});
                         telemetryText += "</span>";
                     }
 
+                    let motorVoltage;
                     if (MOTOR_CONFIG.use_esc_sensor) {
 
                         let escTemperature = MOTOR_TELEMETRY_DATA.temperature[i];
 
                         telemetryText += "<br>";
-                        escTemperature = escTemperature.toString().padStart(MAX_VALUE_SIZE);
+                        escTemperature = escTemperature.toString().padStart(self.MAX_TELEMETRY_VALUE_SIZE);
                         telemetryText += i18n.getMessage('motorsESCTemperature', {motorsESCTempValue: escTemperature});
+
+                        motorVoltage = MOTOR_TELEMETRY_DATA.voltage[i];
+                    } else {
+                        motorVoltage = ANALOG.voltage * (MOTOR_DATA[i] - rangeMin) / full_block_scale;
                     }
 
+                    // Add KV value
+                    let kvValue = motorVoltage > 0 ? MOTOR_TELEMETRY_DATA.rpm[i] / motorVoltage : 0;
```

Comment:
This will need throttle scaling.  Maybe something like below?

let kvValue = motorVoltage > 0 ? MOTOR_TELEMETRY_DATA.rpm[i] / (motorVoltage * (MOTOR_DATA[i] - 1000) / 1000) : 0;

Reply:
The throttle scaling is calculated in the motorVoltage variable, just before.

Reply:
Ah, didnt see that.

motorVoltage will still need to be scaled for throttle when using MOTOR_TELEMETRY_DATA.

I dont think @AlessandroAU  comment previously about the ESC/Motor not getting full voltage is correct. The throttle scaling is more about correcting for duty cycle.

Reply:
We are doing the throttle scaling. In the configurator version also in the firmware driven version.

Reply:
It the same thing, either scale voltage by voltage*throttle or rpm by rpm/throttle. MOTOR_TELEMETRY_DATA.voltage needs to be handled the same as the analogue reading, doesn't look like that's the case?

Reply:
Not in theory, the MOTOR_TELEMETRY_DATA.voltage is the exact voltage reported by the ESC, so we can suppose it is the real voltage and does not need any conversion. But it needs to have the additional cable for the ESC_SENSOR. I don't have any build here with it, so if someone have one and could verify it will be great.

Reply:
No, the telemetry voltage is not the voltage applied to the motor its just vbat again like the ADC on the FC, it will need conversion.

Reply:
By conversion I mean, scaling by the throttle, did you mean converting from the ADC reading to the actual voltage? If so I completely agree.